cpp/daal/src/data_management/train_test_split.cpp

```diff
@@ -39,6 +40,59 @@ typedef daal::data_management::NumericTable::StorageLayout NTLayout;
 const size_t BLOCK_CONST      = 2048;
 const size_t THREADING_BORDER = 8388608;
 
+template <typename IdxType, daal::CpuType cpu>
+services::Status generateShuffledIndicesImpl(const NumericTablePtr & idxTable, const unsigned int seed)
+{
+    const size_t n = idxTable->getNumberOfRows();
+    daal::internal::WriteColumns<IdxType, cpu> idxBlock(*idxTable, 0, 0, n);
+    IdxType * idx = idxBlock.get();
+    DAAL_CHECK_MALLOC(idx);
+
+    daal::services::internal::TArray<IdxType, cpu> swapIdxArr(n);
+    IdxType * swapIdx = swapIdxArr.get();
+    DAAL_CHECK_MALLOC(swapIdx);
+
+    const size_t blockSize = 16 * BLOCK_CONST;
+    const size_t nBlocks   = n / blockSize + !!(n % blockSize);
+
+    daal::conditional_threader_for(n > THREADING_BORDER, nBlocks, [&](size_t iBlock) {
+        const size_t start = iBlock * blockSize;
+        const size_t end   = daal::services::internal::min<cpu, size_t>(start + blockSize, n);
+        daal::internal::BaseRNGs<cpu> baseRng(seed + iBlock);
```

Comment:
not sure that sequential version will reproduce the same results with such approach. (generated numbers will be  different even for number of threads changes)

please use `skipAhead` method. Details can be discussed in mail.

Reply:
I made a mistake, seems as `nBlocks` doesn't depends on number of threads we will see similar sequence of generated numbers.
But we should use seed which was provided by user and do not change it.
So `skipAhead` or `leapfrog` methods are still applicable.

Reply:
Fixed and changed to skipAhead method.